File: app/src/main/java/shaishav/com/bebetter/Data/MySQLiteHelper.java
package shaishav.com.bebetter.Data;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import shaishav.com.bebetter.Data.contracts.LessonContract;

public class MySQLiteHelper extends SQLiteOpenHelper {


    public static final String TABLE_IDEAS = "ideas";
    public static final String COLUMN_ID = "_id";
    public static final String COLUMN_TITLE = "title";
    public static final String COLUMN_LESSON = "lesson";
    public static final String COLUMN_CATEGORY = "categories";
    public static final String COLUMN_CREATED_AT = "createdAt";
    public static final String COLUMN_SERVER_ID = "server_id";
    public static final String COLUMN_IS_PUBLIC="public";

    public static final String TABLE_USAGE = "usage";
    public static final String COLUMN_DATE = "date";
    public static final String COLUMN_USAGE="usage";

    public static final String TABLE_GOAL = "goal";
    public static final String COLUMN_GOAL = "goal";

    private static final String DATABASE_NAME = "bebetter.db";
    // TODO: Set db version to 1
    private static final int DATABASE_VERSION = 1;


    private static final String CREATE_USAGE_STATS = "create table "
            + TABLE_USAGE + "(" + COLUMN_ID
            + " integer primary key autoincrement, " + COLUMN_DATE
            + " long not null,"+COLUMN_USAGE+" long not null, "+COLUMN_SERVER_ID+" text default 'NA');";

    private static final String CREATE_GOAL = "create table "
            + TABLE_GOAL + "(" + COLUMN_ID
            + " integer primary key autoincrement, " + COLUMN_DATE
            + " long not null,"+COLUMN_GOAL+" long not null,"
            +COLUMN_SERVER_ID+" text default 'NA');";


    public MySQLiteHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase database) {
        database.execSQL(LessonContract.LESSON_CREATE_TABLE);
        database.execSQL(CREATE_USAGE_STATS);
        database.execSQL(CREATE_GOAL);

    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        // TODO: comment this out
//        if(oldVersion<newVersion){
//            final String ALTER_TBL = "UPDATE "+MySQLiteHelper.TABLE_USAGE+" SET "
//                    +COLUMN_SERVER_ID+" = 'NA';";
//            db.execSQL(ALTER_TBL);
            //db.execSQL(ORDER_DATABASE_CREATE);
//        }
    }

}

File: app/src/main/java/shaishav/com/bebetter/data/MySQLiteHelper.kt
/*
 * Copyright (c) 2018 Shaishav Gandhi
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  You may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions
 *  and limitations under the License.
 */

package shaishav.com.bebetter.data

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import shaishav.com.bebetter.data.contracts.GoalContract
import shaishav.com.bebetter.data.contracts.PointContract
import shaishav.com.bebetter.data.contracts.UsageContract

/**
 * Created by shaishav.gandhi on 2/27/18.
 */
class MySQLiteHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

  override fun onCreate(database: SQLiteDatabase) {
    database.execSQL(UsageContract.CREATE_USAGE_TABLE)
    database.execSQL(GoalContract.CREATE_GOAL)
    database.execSQL(PointContract.CREATE_POINTS)
  }

  override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {

  }

  companion object {


    private val DATABASE_NAME = "bebetter.db"
    private val DATABASE_VERSION = 1
  }

}